code/modules/vehicles/mecha/mech_bay.dm

```diff
@@ -128,23 +196,37 @@
 		return data
 
 	data["recharge_port"] = list("mech" = null)
-	var/obj/vehicle/sealed/mecha/recharging_mech = recharge_port.recharging_mech_ref?.resolve()
 
-	if(!recharging_mech)
+	var/obj/recharging_machine = recharge_port?.recharging_machine_ref?.resolve()
+	if(!recharging_machine)
 		return data
-	data["recharge_port"]["mech"] = list("health" = recharging_mech.get_integrity(), "maxhealth" = recharging_mech.max_integrity, "cell" = null, "name" = recharging_mech.name,)
 
-	if(QDELETED(recharging_mech.cell))
+	var/obj/item/stock_parts/cell/electron_cache = recharge_port?.recharging_powercell_ref?.resolve()
+	var/obj/vehicle/sealed/mecha/mech = recharging_machine
+	var/list/mechdata = list()
+	if(istype(mech))
+		mechdata["health"] = mech.get_integrity()
+		mechdata["maxhealth"] = mech.max_integrity
+		mechdata["name"] = mech.name
+	else
+		mechdata["health"] = 1
+		mechdata["maxhealth"] = 1
```

Comment:
I think it'd be cute if it read out the machine's integrity as well but that's up to you